rfcs/0132-meson-builds-nix.md

```diff
@@ -0,0 +1,124 @@
+---
+feature: meson_builds_nix
+start-date: 2022-08-25
+author: Anderson Torres
+co-authors: @p01arst0rm
+shepherd-team:
+shepherd-leader:
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+Use meson as an alternative build system for the reference implementation of Nix.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, the reference implementation of Nix evaluator and its companion toolset generated from the Nix source tree are built using the typical `./configure` shell script that relies on autoconf and the standard GNU Make utility.
+
+This build system became clunky and plastered, and consequently hard to understand, modify, improve and port to other systems besides Linux.
```

Comment:
Consider rewording 

"Over time, this buildsystem has been patched and expanded to keep up with the development needs of Nix. However, it has reached a state where the buildsystem is hard to understand or modify, consequently making improvements to the buildsystem difficult.

Additionally, as this current buildsystem has been expanded, changes have been introduced that impact compatibility non GNU/Linux platforms. These issues can hinder development on other platforms (such as *BSD, macOS, or WSL2). In light of these issues, we propose a novel, from-scratch alternative to the current buildsystem infrastructure."